lsp-semantic-tokens.el

```diff
@@ -541,6 +577,144 @@ LOUDLY will be forwarded to OLD-FONTIFY-REGION as-is."
+        (when next-beg
+          (setq next-beg (prop-match-beginning next-beg))
+          ;; Check whether this is the actual matching beginning of next-end
+          (setq tmp (text-property-search-forward 'lsp-semantic-token--comment-end))
+          (when (or (not tmp) (not (equal next-end (prop-match-end tmp))))
+            (setq next-beg nil)))
+
+        (if next-beg
+            (when (>= next-beg end)
+              ;; Does not overlap with (beg end)
+              (setq next-end nil))
+          (lsp-semantic-tokens--remove-comment-syntax-strict next-end next-end)
+          (setq next-end nil))))
+    (cons prev-beg next-end)))
+
+(defun lsp-semantic-tokens--remove-comment-syntax-strict (beg end)
+  "Remove all commnet syntax strictly in (BEG END), even if they
+overlap out of the range."
+  (lsp-save-restriction-and-excursion
+    (with-silent-modifications
+      ;; Remove comment starters
+      (goto-char beg)
+      (cl-do ((loc (text-property-search-forward
+                    'lsp-semantic-token--comment-beg)
+                   (text-property-search-forward
+                    'lsp-semantic-token--comment-beg)))
+          ((or (not loc) (>= (point) end)))
+        (let ((beg-match (prop-match-beginning loc))
+              (end-match (prop-match-end loc)))
+          (remove-text-properties beg-match end-match '(lsp-semantic-token--comment-beg))
+          (cl-loop for i from beg-match below end-match do
+                   (put-text-property i (1+ i) 'syntax-table
+                                      (get-text-property i 'lsp-semantic-token--previous-syntax-table))
+                   (remove-text-properties i (1+ i) '(lsp-semantic-token--previous-syntax-table)))))
+      ;; Remove comment ends
+      (goto-char end)
+      (cl-do ((loc (text-property-search-backward
+                    'lsp-semantic-token--comment-end)
+                   (text-property-search-backward
+                    'lsp-semantic-token--comment-end)))
+          ((or (not loc) (<= (point) beg)))
+        (let ((beg-match (prop-match-beginning loc))
+              (end-match (prop-match-end loc)))
+          (remove-text-properties beg-match end-match '(lsp-semantic-token--comment-end))
+          (cl-loop for i from beg-match below end-match do
+                   (put-text-property i (1+ i) 'syntax-table
+                                      (get-text-property i 'lsp-semantic-token--previous-syntax-table))
+                   (remove-text-properties i (1+ i) '(lsp-semantic-token--previous-syntax-table))))))))
+
+(defun lsp-semantic-tokens--put-comment-syntax (beg end)
+  "Set the comment syntax from BEG to END."
+  (let* ((overlapping (lsp-semantic-tokens--get-overlapping-comments beg end))
+         (new-beg (car overlapping))
+         (new-end (cdr overlapping)))
+    (when new-beg
+      (setq beg new-beg))
+    (when new-end
+      (setq end new-end)))
+  (lsp-semantic-tokens--remove-comment-syntax-strict beg end)
```

Comment:
given that `lsp--semantic-tokens-fontify` removes comment syntax right at the outset, this call to `lsp-semantic-tokens--remove-comment-syntax-strict` might needlessly degrade performance? Or is this call necessary for some reason I don't see right now?